packages/core/src/future.ts

```diff
@@ -27,10 +27,9 @@ export interface NewFuture<T> {
   resolve(value: Value<T>): void;
 }
 
-export function createFuture<T>(): NewFuture<T> {
+export function createFuture<T>(result?: Value<T>): NewFuture<T> {
   let runLoop = createRunLoop();
   let consumers: Consumer<T>[] = [];
-  let result: Value<T>;
```

Comment:
What about following the promise convension and having a `Future.resolve()` method.

Reply:
That would make sense too, I don't have a strong opinion on this either way. I was a bit unsure of how to name such a thing, so I just ended up picking this solution since it seemed to be less of a hassle.

Reply:
Yeah, it's just that `Future` and `createFuture` are part of our public API, and so as we converge on 2.0 stable, we need to be extra careful. I think that `Future.resolve(x)` is very straightforward and seems 99.9% likely to be part of the public API going forward, but `createFuture({type: 'completed', value: x }).future` seems like it will be rarely used except internally, takes up the real estate of the a potential argument to `createFuture()` (no idea what that might be)

Reply:
We could also add this as a low level API and add helper functions such as `createResolvedFuture` or something? Not sure how I feel about `Future.resolve` as an API, it matches the promise API, but not any other Effection APIs.